cmd/block-generator/generator/generate.go

```diff
@@ -42,13 +42,21 @@ const (
 	assetDestroy TxTypeID = "asset_destroy"
 
 	assetTotal = uint64(100000000000000000)
-	fee        = uint64(1000)
-)
 
-const (
 	consensusTimeMilli int64 = 4500
 )
 
+// consensus parameters.
+var (
+	minBal uint64
+	fee    uint64
+)
+
+func init() {
+	minBal = config.Consensus[protocol.ConsensusFuture].MinBalance
```

Comment:
it might be better to move this to `generatePaymentTxnInternal()` where `g.protocol` can be accessed

Reply:
I moved fee to be a constant to avoid referencing the consensus params in those places, and took this suggestion for min balance. Its tempting to take it all out and just let indexer crash with the overspend error, since generator doesn't do anything special in that case now.

Reply:
It's just "future" is specified in 3 places now, which is not optimal. I would add consensus params struct to the generator struct, so that all parameters can easily be referenced.

Reply:
good idea, added the params to the generator struct.